### PR TITLE
Add latency send/recieve which can be done async with new response type

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: metrics/latency.go [hunk 1]
```diff
@@ -82,3 +82,120 @@ func ProbeLatency(ch channel2.Channel, histogram Histogram, interval time.Durati
...
+
+	lastSend := info.NowInMilliseconds()
+	for {
+		time.Sleep(self.interval)
+		if self.ch.IsClosed() {
+			return
+		}
+
+		now := info.NowInMilliseconds()
+		if now-lastSend > 10000 {
```
Comment (michaelquigley) [inline metrics/latency.go:134]: Can we make this configurable? Probably should fix up any hard-coded timeouts anywhere we run across them (especially in new code).

File: metrics/latency.go [hunk 2]
```diff
@@ -82,3 +82,120 @@ func ProbeLatency(ch channel2.Channel, histogram Histogram, interval time.Durati
...
+		}
+	}
+}
+
+func (self *LatencyResponder) responseSender() {
+	log := pfxlog.ContextLogger(self.ch.Label())
+	log.Debug("started")
+	defer log.Debug("exited")
+
+	ticker := time.NewTicker(10 * time.Second)
```
Comment (michaelquigley) [inline metrics/latency.go:183]: Same here.

**Review by michaelquigley — APPROVED** [review]

**Comment by ekoby** [comment]
this change does not seem backward compatible: how does the peer determine what to expect as the response to latency probe?

File: metrics/latency.go (hunk 1 above)
Comment (plorenz) [reply metrics/latency.go:134]: I just removed this check. I had copied this from the original latency code, but really, there's no reason to double check the latency interval that's getting passed in. The value that's passed in is settable via config.

**Comment by plorenz** [comment]
> this change does not seem backward compatible: how does the peer determine what to expect as the response to latency probe?
> 
> SDK will need to support both types of latency responses (per ziti network, or even per Edge Router)

This check is going to be used on the control channel only, which didn't have latency checks before, so we shouldn't have to worry about backwards compatibility there

**Pull request merged**